xarray/coding/cftimeindex.py

```diff
@@ -254,3 +299,28 @@ def __contains__(self, key):
     def contains(self, key):
         """Needed for .loc based partial-string indexing"""
         return self.__contains__(key)
+
+    def __unicode__(self):
+        """Return a string representation for this object.
+
+        Adds a calendar attribute to denote the calendar type of the index.
+        Adapted from pandas.core.indexes.base.__unicode__
+        """
+        klass = self.__class__.__name__
+        data = self._format_data()
```

Comment:
This is a little dangerous: we're using private methods here which might go away without notice. I have been burned by this in the past (specifically in xarray with pandas).

The other reason to hold off on copying the pandas repr() is that we don't implement casting like `pd.to_datetime()` instead the CFTimeIndex constructor, e.g.,
```
In [2]: pd.DatetimeIndex(['2000', '2001'])
Out[2]: DatetimeIndex(['2000-01-01', '2001-01-01'], dtype='datetime64[ns]', freq=None)
```

(to be clear, I don't think this is a good idea either, but at least it ensures that the repr roundtrips faithfully)